src/libcsg/modules/io/lammpsdatareader.cc

```diff
@@ -88,7 +88,7 @@ bool LAMMPSDataReader::ReadTopology(string file, Topology &top) {
   NextFrame(top);
 
   fl_.close();
-  for (Molecule *mol : top.Molecules()) {
+  for (auto &mol : top.Molecules()) {
```

Comment:
why this change? `auto`does not add anything for me.

Reply:
Ya, I've begun to swing towards using auto more, specifically in cases where the type is really long and 2 if I were to change from using a unique_ptr to a smart pointer I would have to update each and every one of these for loops. It is way more work even if it is less obvious.